### PR TITLE
[Merged by Bors] - chore(List): delete old, unused, deprecated theorems

### PR DESCRIPTION
**Pull request by kim-em**
These theorems have been deprecated for at least a year, and are unused.

---

- [x] depends on: #12171

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/BigOperators/Basic.lean [hunk 1]
```diff
@@ -215,58 +215,11 @@ theorem map_prod [CommMonoid β] [CommMonoid γ] {G : Type*} [MonoidHomClass G 
...
 #align monoid_hom.map_prod map_prodₓ
 #align add_monoid_hom.map_sum map_sumₓ
 #align mul_equiv.map_prod map_prodₓ
 #align add_equiv.map_sum map_sumₓ
 
-@[deprecated _root_.map_list_prod]
-protected theorem RingHom.map_list_prod [Semiring β] [Semiring γ] (f : β →+* γ) (l : List β) :
-    f l.prod = (l.map f).prod :=
-  map_list_prod f l
-#align ring_hom.map_list_prod RingHom.map_list_prod
```
Comment (urkud) [inline Mathlib/Algebra/BigOperators/Basic.lean:229]: IMHO, we should realign these lemmas to the generic lemmas in the root namespace.

File: Mathlib/Algebra/GeomSum.lean [hunk 2]
```diff
@@ -402,13 +402,14 @@ theorem geom_sum_inv [DivisionRing α] {x : α} (hx1 : x ≠ 1) (hx0 : x ≠ 0)
 
 variable {β : Type*}
 
+-- TODO: for consistency, the next two lemmas should be moved to the root namespace
```
Comment (urkud) [inline Mathlib/Algebra/GeomSum.lean:405]: .. and generalized to `RingHomClass`

Comment (urkud) [reply Mathlib/Algebra/GeomSum.lean:405]: Otherwise LGTM. Thanks!
bors d+

File: Mathlib/Algebra/BigOperators/Basic.lean (hunk 1 above)
Comment (digama0) [inline Mathlib/Algebra/BigOperators/Basic.lean:229]: Don't delete `#align`s, make them `#noalign` or (ideally) point to the deprecation target instead

Comment (urkud) [reply Mathlib/Algebra/BigOperators/Basic.lean:229]: I guess, this mean
bors d-

**Comment by semorrison** [comment]
This has thoroughly rotted, oh well. I am really looking forward to #align going away.

File: Mathlib/Data/List/Basic.lean [hunk 3]
```diff
@@ -1460,16 +1442,6 @@ theorem eq_cons_of_length_one {l : List α} (h : l.length = 1) :
   omega
 #align list.eq_cons_of_length_one List.eq_cons_of_length_one
 
-@[deprecated get_eq_iff] -- 2023-01-05
```
Comment (grunweg) [inline Mathlib/Data/List/Basic.lean:1463]: The aligns moved a few hundred lines up.

**Review by grunweg — APPROVED** [review]
Looks straightforward; thank you for doing this.

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**